core/trino-main/src/main/java/io/trino/operator/scalar/CharacterStringCasts.java

```diff
@@ -121,6 +121,24 @@ public static Slice varcharToCharSaturatedFloorCast(@LiteralParameter("y") long
         return codePointsToSliceUtf8(codePoints);
     }
 
+    @ScalarOperator(OperatorType.SATURATED_FLOOR_CAST)
+    @SqlType("varchar(y)")
+    @LiteralParameters({"x", "y"})
+    public static Slice varcharToVarcharSaturatedFloorCast(@LiteralParameter("y") long y, @SqlType("varchar(x)") Slice slice)
+    {
```

Comment:
```
verify(y  < x, "...");
```

?

Reply:
varchar -> char cast truncates it so I guess we should be compatible